fv3core/stencils/del2cubed.py

```diff
@@ -73,47 +70,98 @@ def corner_fill(q: FloatField):
             q = q[0, -1, 0]
 
 
-def compute(qdel: FloatField, nmax: int, cd: float, km: int):
-    grid = spec.grid
-    origin = (grid.isd, grid.jsd, 0)
-
-    # Construct some necessary temporary storage objects
-    fx = utils.make_storage_from_shape(
-        qdel.shape, origin=origin, cache_key="del2cubed_fx"
-    )
-    fy = utils.make_storage_from_shape(
-        qdel.shape, origin=origin, cache_key="del2cubed_fy"
-    )
-
-    # set up the temporal loop
-    ntimes = min(3, nmax)
-    for n in range(1, ntimes + 1):
-        nt = ntimes - n
-        origin = (grid.is_ - nt, grid.js - nt, 0)
+class HyperdiffusionDamping:
+    """
+    Fortran name is del2cubed
+    """
+
+    def __init__(self, grid, qdel: FloatField, km: int):
+        """
+        Args:
+            grid: fv3core grid object
+            qdel (inout): Variable to be filterd
+            km: Number of vertical levels (grid.npz)
+        """
+        self.grid = spec.grid
+        self.origin = (self.grid.isd, self.grid.jsd, 0)
+        self.domain = (self.grid.nid, self.grid.njd, km)
+        ax_offsets = axis_offsets(spec.grid, self.origin, self.domain)
+        self._fx = utils.make_storage_from_shape(
+            qdel.shape, origin=self.origin, cache_key="del2cubed_fx"
+        )
+        self._fy = utils.make_storage_from_shape(
+            qdel.shape, origin=self.origin, cache_key="del2cubed_fy"
+        )
 
-        # Fill in appropriate corner values
-        corner_fill(
-            qdel, origin=(grid.isd, grid.jsd, 0), domain=(grid.nid, grid.njd, km)
+        self._corner_fill = StencilWrapper(
+            func=corner_fill,
+            externals={
+                **ax_offsets,
+            },
+            origin=self.origin,
+            domain=self.domain,
         )
+        self._compute_zonal_flux = StencilWrapper(func=compute_zonal_flux)
+        self._compute_meridional_flux = StencilWrapper(func=compute_meridional_flux)
+        self._update_q = StencilWrapper(func=update_q)
+
+    def __call__(self, qdel: FloatField, nmax: int, cd: float, km: int):
+        """
+        Perform hyperdiffusion damping/filtering
+
+        Args:
+            qdel (inout): Variable to be filterd
+            nmax: Number of times to apply filtering
+            cd: Damping coeffcient
+            km: Number of vertical levels (grid.npz)
```

Comment:
km doesn't need to both be a runtime argument and an init argument -- you can remove km from here and use self._km